optuna/importance/_mean_decrease_impurity.py

```diff
@@ -0,0 +1,136 @@
+
+    _available = True
+except ImportError as e:
+    _import_error = e
+    _available = False
+
+
+@experimental("1.5.0")
+class MeanDecreaseImpurityImportanceEvaluator(BaseImportanceEvaluator):
+    """Mean Decrease Impurity (MDI) parameter importance evaluator.
+
+    This evaluator fits a random forest that predicts objective values given hyperparameter
+    configurations. Feature importances are then computed using MDI.
+
+    .. note::
+
+        Requires the `sklean <https://scikit-learn.org/stable/>`_ Python package.
+
+    Args:
+        n_estimators:
+            Number of trees in the random forest.
+        max_depth:
+            The maximum depth of each tree in the random forest.
+        random_seed:
+            Seed for the random forest.
+    """
+
+    def __init__(
+        self, n_estimators: int = 16, max_depth: int = 64, random_state: Optional[int] = None
+    ) -> None:
+        _check_sklearn_availability()
+
+        self._forest = RandomForestRegressor(
+            n_estimators=n_estimators,
+            max_depth=max_depth,
+            min_samples_split=2,
+            min_samples_leaf=1,
+            random_state=random_state,
+        )
+
+    def evaluate(self, study: Study, params: Optional[List[str]]) -> Dict[str, float]:
+        distributions = _get_distributions(study, params)
+        params_data, values_data = _get_study_data(study, distributions)
+
+        if params_data.size == 0:  # `params` were given but as an empty list.
+            return OrderedDict()
+
+        params_data, cols_to_raw_cols = _encode_categorical(params_data, distributions.values())
+
+        forest = self._forest
+        forest.fit(params_data, values_data)
+        feature_importances = forest.feature_importances_
+        feature_importances_reduced = numpy.zeros((len(distributions),), dtype=numpy.float32)
+        numpy.add.at(feature_importances_reduced, cols_to_raw_cols, feature_importances)
+
+        param_importances = OrderedDict()
+        param_names = list(distributions.keys())
+        for i in feature_importances_reduced.argsort()[::-1]:
+            param_importances[param_names[i]] = feature_importances_reduced[i].item()
+
+        return param_importances
+
+
+def _encode_categorical(
+    params_data: numpy.ndarray, distributions: ValuesView[BaseDistribution],
+) -> Tuple[numpy.ndarray, numpy.ndarray]:
+    # Transform the `params_data` matrix by expanding categorical integer-valued columns to one-hot
+    # encoding matrices. Note that the resulting matrix can be sparse and potetially very big.
+
+    numerical_cols = []
+    categorical_cols = []
+    categories = []
+
+    for col, distribution in enumerate(distributions):
+        if isinstance(distribution, CategoricalDistribution):
+            categories.append(numpy.arange(len(distribution.choices)))
+            categorical_cols.append(col)
+        else:
+            numerical_cols.append(col)
+    assert col == params_data.shape[1] - 1
```

Comment:
Sorry could you elaborate a bit? It seems to me like a valid assertion since `params_data` is passed by the caller (an optuna-internal) to this function, and this assertion doesn't overlap with any previous assertion within the scope of this function.

Reply:
At first, I thought this assertion could be omitted, but for the sake of readability, I think it should be left out. Sorry for the confusion.